keras_core/backend/torch/core.py

```diff
@@ -114,32 +106,30 @@ def __eq__(self, other):
 
 
 def convert_to_tensor(x, dtype=None):
-    dtype = to_torch_dtype(dtype or getattr(x, "dtype", None))
-    device = get_device()
-    if isinstance(x, int):
-        dtype = torch.int32
-    if isinstance(x, float):
-        dtype = torch.float32
+    if is_tensor(x):
+        return x
```

Comment:
Will have to check dtype and convert to the said dtype if dtype is passed. Tests ae failing when we pass integer arguments with dtype boolean

Reply:
Yes, let's add the check and conversion here.